hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -991,10 +992,6 @@ public void update(HoodieRollbackMetadata rollbackMetadata, String instantTime)
       // Find the deltacommits since the last compaction
       Option<Pair<HoodieTimeline, HoodieInstant>> deltaCommitsInfo =
           CompactionUtils.getDeltaCommitsSinceLatestCompaction(metadataMetaClient.getActiveTimeline());
-      if (!deltaCommitsInfo.isPresent() || deltaCommitsInfo.get().getKey().empty()) {
```

Comment:
why remove this?

Reply:
here irrespective of whether the commit to rollback is already present in MDT or not, we might have to go ahead and add the delta commit. thats why I had to fix this.